pkg/gitreceive/build.go

```diff
@@ -81,11 +82,14 @@ func build(conf *Config, builderKey, gitSha string) error {
 	buildDir := filepath.Join(repoDir, "build")
 
 	slugName := fmt.Sprintf("%s:git-%s", appName, shortSha)
-	imageName := strings.Replace(slugName, ":", "-", -1)
 	if err := os.MkdirAll(buildDir, os.ModeDir); err != nil {
 		return fmt.Errorf("making the build directory %s (%s)", buildDir, err)
 	}
-	tmpDir := os.TempDir()
+
+	tmpDir, err := ioutil.TempDir(buildDir, "tmp")
```

Comment:
Every git push requires a different TMP dir or else there will be a conflict of files between each versions or commit unless we delete TMP DIR for each build